File: day1.py
from functools import reduce
from sys import stdin


def update(calories, line):
    if line:
        return [calories[0] + int(line)] + calories[1:]
    return [0] + calories


def main():
    lines = [line.strip() for line in stdin.readlines()]

    calories = sorted(reduce(update, lines, [0]), reverse=True)

    print(f"max: {calories[0]}")
    print(f"top three: {sum(calories[i] for i in range(3))}")


__name__ == '__main__' and main()

File: day1.hs
import Data.List

update :: [Int] -> String -> [Int]
update [] _ = []
update xs [] = 0 : xs
update (x : xs) s = (x + read s :: Int) : xs

main = do
  res <- reverse . sort . foldl update [0] . lines <$> getContents
  print $ head res
  print $ sum $ take 3 res
